### PR TITLE
SMP heartbeat to maintain the connection

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Protocol.hs [hunk 1]
```diff
@@ -71,11 +71,13 @@ data Command (a :: Party) where
   OFF :: Command Recipient
   DEL :: Command Recipient
   SEND :: MsgBody -> Command Sender
+  PING :: Command Sender
```
Comment (efim-poberezkin) [inline src/Simplex/Messaging/Protocol.hs:74]: not exclusive to sender? well I guess that's just previous SMP abstraction being stretched here.. whatever

but maybe add a separate `Agent` party?

Comment (efim-poberezkin) [reply src/Simplex/Messaging/Protocol.hs:74]: also could you separate `Broker` and "Agent" commands with comment headers please? ("Agent" commands are of `Recipient` & `Sender`, `Agent` if you add it as party. `Recipient` and `Sender` could be separate sections)

File: src/Simplex/Messaging/Crypto.hs [hunk 2]
```diff
@@ -243,3 +245,13 @@ rsaPrivateKey pk =
       R.private_dQ = undefined,
       R.private_qinv = undefined
     }
+
```
Comment (efim-poberezkin) [inline src/Simplex/Messaging/Crypto.hs:248]: this file probably got into pr by a mistake

**Review by efim-poberezkin — APPROVED** [review]

File: src/Simplex/Messaging/Protocol.hs (hunk 1 above)
Comment (epoberezkin) [reply src/Simplex/Messaging/Protocol.hs:74]: no problem if it is assigned to Sender. We do not create one TCP connection per queue, meaning that in one. TCP connection there can be both "Sender" and "Receiver" commands.

The point to make it belong to Sender is that if/when we add authentication, the server would reject Recipient commands without server user authentication.

File: src/Simplex/Messaging/Crypto.hs (hunk 2 above)
Comment (epoberezkin) [reply src/Simplex/Messaging/Crypto.hs:248]: just didn't want to delete this function - let it be :)

File: src/Simplex/Messaging/Protocol.hs [hunk 3]
```diff
@@ -64,14 +64,17 @@ type SenderId = QueueId
 type QueueId = Encoded
 
 data Command (a :: Party) where
+  -- Agent Recipient commands
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Protocol.hs:67]: these are not agent commands

**Pull request merged**